packages/core/src/common/keybinding.ts

```diff
@@ -248,4 +316,42 @@ export class KeybindingRegistry {
     isPseudoCommand(commandId: string): boolean {
         return commandId === KeybindingRegistry.PASSTHROUGH_PSEUDO_COMMAND;
     }
+
+    setKeymap(rawKeyBindings: RawKeybinding[]) {
+        const customBindings: Keybinding[] = [];
+        for (const rawKeyBinding of rawKeyBindings) {
+            if (this.commandRegistry.getCommand(rawKeyBinding.command)) {
+                // TODO check if valid context? valid keybinding?
+                // Create Keycode from string?
+                // Create new Keybinding and assign it
+                // If one keybinding is wrong, reset to the default ones.
+
+                const code = TheiaKeyCodeUtils.parseKeystroke(rawKeyBinding.keybinding);
+                if (code) {
+                    let context: KeybindingContext | undefined;
+                    if (rawKeyBinding.context) {
+                        context = this.contextRegistry.getContext(rawKeyBinding.context);
+                    }
+
+                    customBindings.push({
+                        commandId: rawKeyBinding.command,
+                        keyCode: code,
+                        contextId: context ? context.id : undefined
+                    });
+
+                } else {
+                    this.resetKeybindings();
```

Comment:
If you want to restore, then please to the previous state and not to 'no keymaps'.

Reply:
But if the json file is invalid, I'd rather have the default keybindings than the previous keybindings no? I went this way as VScode does it like this too. Wdyt?